go/client/cmd_chat_send.go

```diff
@@ -74,83 +77,134 @@ func (c *cmdChatSend) Run() (err error) {
 		conversationInfo = *resolved
 	}
 
-	switch {
-	case userChosen && len(c.message) == 0:
-		c.message, err = c.G().UI.GetTerminalUI().Prompt(PromptDescriptorEnterChatMessage, fmt.Sprintf("Send to [%s]? Hit Ctrl-C to cancel, or enter message content to send: ", conversationInfo.TlfName))
-		if err != nil {
-			return err
-		}
-	case userChosen:
-		return errors.New("potential command line argument parsing error: we had a message before letting user choose a conversation")
-	case len(c.message) == 0:
-		c.message, err = c.G().UI.GetTerminalUI().Prompt(PromptDescriptorEnterChatMessage, "Please enter message content: ")
-		if err != nil {
-			return err
-		}
-	default:
-	}
-
 	var args chat1.PostLocalArg
 	args.ConversationID = conversationInfo.Id
 
 	var msgV1 chat1.MessagePlaintextV1
 	// msgV1.ClientHeader.{Sender,SenderDevice} are filled by service
 	msgV1.ClientHeader.Conv = conversationInfo.Triple
 	msgV1.ClientHeader.TlfName = conversationInfo.TlfName
-	msgV1.ClientHeader.MessageType = chat1.MessageType_TEXT
-	msgV1.MessageBody = chat1.NewMessageBodyWithText(chat1.MessageText{Body: c.message})
-
-	args.MessagePlaintext = chat1.NewMessagePlaintextWithV1(msgV1)
 
-	if _, err = chatClient.PostLocal(ctx, args); err != nil {
-		return err
-	}
-
-	if len(c.setTopicName) > 0 {
+	// Do one of set topic name, set headline, send message
+	switch {
+	case c.setTopicName != "":
 		if conversationInfo.Triple.TopicType == chat1.TopicType_CHAT {
 			c.G().UI.GetTerminalUI().Printf("We are not supporting setting topic name for chat conversations yet. Ignoring --set-topic-name >.<")
+			return nil
 		}
 		msgV1.ClientHeader.MessageType = chat1.MessageType_METADATA
 		msgV1.MessageBody = chat1.NewMessageBodyWithMetadata(chat1.MessageConversationMetadata{ConversationTitle: c.setTopicName})
-		args.MessagePlaintext = chat1.NewMessagePlaintextWithV1(msgV1)
-		if _, err := chatClient.PostLocal(ctx, args); err != nil {
-			return err
+	case c.setHeadline != "":
+		msgV1.ClientHeader.MessageType = chat1.MessageType_HEADLINE
+		msgV1.MessageBody = chat1.NewMessageBodyWithHeadline(chat1.MessageHeadline{Headline: c.setHeadline})
+	case c.clearHeadline:
+		msgV1.ClientHeader.MessageType = chat1.MessageType_HEADLINE
+		msgV1.MessageBody = chat1.NewMessageBodyWithHeadline(chat1.MessageHeadline{Headline: ""})
+	default:
+		// Ask for message contents
+		if len(c.message) == 0 {
+			promptText := "Please enter message content: "
+			if userChosen {
```

Comment:
I agree that sending a message, `setHeadline`, `clearHeadline`, and `setTopicName` should act the same in this respect.
When sending a message there are 3 ways to provide the message:
1. in the command
2. with --stdin
3. when prompted
It seems like we only prompted in case 3 and then the prompt text depended on whether `userChosen`.

Do you think we should always confirm when `userChosen`? I figured the prompt was only there because we need to get `c.message` and the different text was just a convenience.